src/Spinner.php

```diff
@@ -71,24 +73,29 @@ public function spin(Closure $callback): mixed
             } else {
                 $result = $callback();
 
-                posix_kill($pid, SIGHUP);
-                pcntl_async_signals($originalAsync);
-                pcntl_signal(SIGINT, SIG_DFL);
-
-                $this->eraseRenderedLines();
-                $this->showCursor();
+                $this->resetTerminal($originalAsync, $pid);
 
                 return $result;
             }
         } catch (\Throwable $e) {
-            $this->showCursor();
-            pcntl_async_signals($originalAsync);
-            pcntl_signal(SIGINT, SIG_DFL);
+            $this->resetTerminal($originalAsync, $pid);
```

Comment:
```suggestion
            $this->resetTerminal($originalAsync, $pid ?? null);
```

Reply:
That's right, thank you. Just updated the PR with the changes :+1: